### PR TITLE
Common dao

### PR DESCRIPTION
**Pull request by Mellifaro**
Created entities, repositories, services. Created some startup tests.

### CODE REVIEW

File: basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/DeviceEntity.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * Created by Виктор on 18.03.2017.
+ */
+@Entity
+@Table(name = "devices")
+public class DeviceEntity implements Serializable{
+    private static final long serialVersionUID = -506229924089552354L;
+
+    @Id
+    @Column(name = "id")
+    private String id;
```
Comment (Dominys) [inline basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/DeviceEntity.java]: @Mellifaro All our entities will use String Id - make some base class for this.

File: basis/common-dao/common-dao-impll/src/main/java/com/softgroup/common/dao/impl/repositories/DeviceRepository.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.softgroup.common.dao.impl.repositories;
+
+import com.softgroup.common.dao.api.entities.DeviceEntity;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+/**
+ * Created by Виктор on 18.03.2017.
+ */
+public interface DeviceRepository extends PagingAndSortingRepository<DeviceEntity, String>{
```
Comment (Dominys) [inline basis/common-dao/common-dao-impll/src/main/java/com/softgroup/common/dao/impl/repositories/DeviceRepository.java]: @Mellifaro All our entities will use String ID. Make some base repository to reduce number of generic parameters.

**Comment by Mellifaro** [comment]
Created base repository for entities

File: basis/common-dao/common-dao-impll/src/main/resources/changelogs/changelog.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext
+   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
+   http://www.liquibase.org/xml/ns/dbchangelog
+   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <changeSet author="Mellifaro" id="changeset1">
+        <createTable tableName="profiles">
+            <column name="id" type="varchar(255)" />
```
Comment (Dominys) [inline basis/common-dao/common-dao-impll/src/main/resources/changelogs/changelog.xml]: @Mellifaro Why 255? Uuid length is 36

File: basis/common-dao/common-dao-impll/src/main/resources/changelogs/changelog.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            <column name="phone_number" type="varchar(255)" />
```
Comment (Dominys) [inline basis/common-dao/common-dao-impll/src/main/resources/changelogs/changelog.xml]: @Mellifaro Where did you see such long phone-numbers ?

**Comment by Dominys** [comment]
@Mellifaro Where is base entity ?

File: basis/common-dao/common-dao-impll/src/main/resources/changelogs/changelog.xml (hunk 3 above)
Comment (Mellifaro) [reply basis/common-dao/common-dao-impll/src/main/resources/changelogs/changelog.xml]: ok. I will change length to 36 chars

File: basis/common-dao/common-dao-impll/src/main/resources/changelogs/changelog.xml (hunk 4 above)
Comment (Mellifaro) [reply basis/common-dao/common-dao-impll/src/main/resources/changelogs/changelog.xml]: I will change length to 20 chars

**Comment by Mellifaro** [comment]
ok. I will create baseEntity

**Comment by Mellifaro** [comment]
Created BaseEntity

**Review by Dominys — APPROVED** [review]

**Pull request merged**